lib/spack/llnl/util/tty/__init__.py

```diff
@@ -65,17 +129,28 @@ def process_stacktrace(countback):
     return st_text
 
 
-def msg(message, *args, **kwargs):
-    newline = kwargs.get('newline', True)
-    st_text = ""
-    if _stacktrace:
-        st_text = process_stacktrace(2)
-    if newline:
-        cprint("@*b{%s==>} %s" % (st_text, cescape(message)))
+def get_timestamp(force=False):
+    """Get a string timestamp"""
+    if _debug or _timestamp or force:
+        return datetime.now().strftime("[%Y-%m-%d-%H:%M:%S.%f] ")
     else:
-        cwrite("@*b{%s==>} %s" % (st_text, cescape(message)))
-    for arg in args:
-        print(indent + str(arg))
+        return ''
+
+
+def msg(message, *args, **kwargs):
+    if msg_enabled():
```

Comment:
minor request: just make this and the others start with (for less overall indentation):

```python
if not msg_enabled():
    return
```